I get many _many_ test failures for `python38Packages.ansible-runner`, citing amongst other things:

```
Unhandled error:
 Traceback (most recent call last):
  File "/nix/store/rlzzry7366ihk1fv50pk92p0g7l2jzav-python3.8-ansible-2.9.27/lib/python3.8/site-packages/ansible/utils/path.py", line 90, in makedirs_safe
    os.makedirs(b_rpath, mode)
  File "/nix/store/8il5qnc86imjwjxvnqqaahdfvilmmd21-python3-3.8.12/lib/python3.8/os.py", line 213, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/nix/store/8il5qnc86imjwjxvnqqaahdfvilmmd21-python3-3.8.12/lib/python3.8/os.py", line 213, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/nix/store/8il5qnc86imjwjxvnqqaahdfvilmmd21-python3-3.8.12/lib/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: b'/homeless-shelter'
```

So it looks like we need to at least set `HOME` to a tempdir before the `checkPhase`

--- a/pkgs/development/python-modules/ansible-runner/default.nix
+++ b/pkgs/development/python-modules/ansible-runner/default.nix
@@ -8,20 +8,33 @@
 , six
 , stdenv
 , ansible
-, pytest
 , mock
+, openssh
+, pytest-mock
+, pytest-timeout
+, pytest-xdist
+, pytestCheckHook
 }:
 
 buildPythonPackage rec {
   pname = "ansible-runner";
-  version = "2.0.2";
+  version = "2.0.3";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "c02b690803ec0be4453411c53743cd3fdca1dfc66dfa075794e14e717c5b61b3";
+    sha256 = "15j0ljgw1rjxq4xiywxxxnxj4r6vwk8dwljkdsjmq3qmwgifcswg";
   };
 
-  checkInputs = [ pytest mock ];
+  checkInputs = [
+    ansible # required to place ansible CLI onto the PATH in tests
+    pytestCheckHook
+    pytest-mock
+    pytest-timeout
+    pytest-xdist
+    mock
+    openssh
+  ];
+
   propagatedBuildInputs = [
     ansible
     psutil
@@ -31,13 +44,30 @@ buildPythonPackage rec {
     six
   ];
 
-  # test_process_isolation_settings is currently broken on Darwin Catalina
-  # https://github.com/ansible/ansible-runner/issues/413
-  checkPhase = ''
-    HOME=$TMPDIR pytest \
-      --ignore test/unit/test_runner.py \
-      -k "not prepare ${lib.optionalString stdenv.isDarwin "and not process_isolation_settings"}"
+  preCheck = ''
+    export HOME=$(mktemp -d)
   '';
+
+  disabledTests = [
+    "test_callback_plugin_task_args_leak" # requires internet
+    "test_env_accuracy"
+    "test_large_stdout_blob" # times out on slower hardware
+  ]
+    # test_process_isolation_settings is currently broken on Darwin Catalina
+    # https://github.com/ansible/ansible-runner/issues/413
+  ++ lib.optional stdenv.isDarwin "process_isolation_settings";
+
+  disabledTestPaths = [
+    # these tests unset PATH and then run executables like `bash` (see https://github.com/ansible/ansible-runner/pull/918)
+    "test/integration/test_runner.py"
+    "test/unit/test_runner.py"
+  ]
+  ++ lib.optionals stdenv.isDarwin [
+    # integration tests on Darwin are not regularly passing in ansible-runner's own CI
+    "test/integration"
+    # these tests write to `/tmp` which is not writable on Darwin
+    "test/unit/config/test__base.py"
+  ];
 
   meta = with lib; {
     description = "Helps when interfacing with Ansible";

